clients/java/src/test/java/io/camunda/zeebe/client/process/MigrateProcessInstanceTest.java

```diff
@@ -63,9 +92,75 @@ public void shouldMigrateWithMigrationPlanBuilder() {
         client.newMigrateProcessInstanceCommand(PI_KEY).migrationPlan(migrationPlan);
 
     // when
-    final ThrowingCallable send = migrateProcessInstanceCommand::send;
+    migrateProcessInstanceCommand.send().join();
+
+    // then
+    final MigrateProcessInstanceRequest request = gatewayService.getLastRequest();
+    assertRequest(request, 2);
+    final MappingInstruction mappingInstruction1 =
+        request.getMigrationPlan().getMappingInstructions(0);
+    assertMappingInstruction(mappingInstruction1, ELEMENT_ID_A, ELEMENT_ID_B);
+    final MappingInstruction mappingInstruction2 =
+        request.getMigrationPlan().getMappingInstructions(1);
+    assertMappingInstruction(mappingInstruction2, ELEMENT_ID_C, ELEMENT_ID_D);
+  }
+
+  @Test
+  public void shouldMigrateWithMigrationPlanBuilderAndAdditionalMappingInstruction() {
+    // given
+    final MigrationPlan migrationPlan =
+        MigrationPlan.newBuilder()
+            .withTargetProcessDefinitionKey(PD_KEY)
+            .addMappingInstruction(ELEMENT_ID_A, ELEMENT_ID_B)
+            .build();
+
+    final MigrateProcessInstanceCommandFinalStep migrateProcessInstanceCommand =
+        client
+            .newMigrateProcessInstanceCommand(PI_KEY)
+            .migrationPlan(migrationPlan)
+            .addMappingInstruction(ELEMENT_ID_C, ELEMENT_ID_D);
+
+    // when
+    migrateProcessInstanceCommand.send().join();
 
     // then
-    assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(send);
+    final MigrateProcessInstanceRequest request = gatewayService.getLastRequest();
+    assertRequest(request, 2);
+    final MappingInstruction mappingInstruction1 =
+        request.getMigrationPlan().getMappingInstructions(0);
+    assertMappingInstruction(mappingInstruction1, ELEMENT_ID_A, ELEMENT_ID_B);
+    final MappingInstruction mappingInstruction2 =
+        request.getMigrationPlan().getMappingInstructions(1);
+    assertMappingInstruction(mappingInstruction2, ELEMENT_ID_C, ELEMENT_ID_D);
+  }
+
+  @Test
+  public void shouldThrowErrorWhenTryToMigrateProcessInstanceWithNullMappingInstruction() {
+    // when
+    Assertions.assertThatThrownBy(
+            () ->
+                client
+                    .newModifyProcessInstanceCommand(PI_KEY)
+                    .activateElement(ELEMENT_ID_A)
+                    .withVariable(null, null)
+                    .send()
+                    .join())
+        .isInstanceOf(IllegalArgumentException.class);
+  }
```

Comment:
❌ This test is testing modification, not migration.

However, when adjusted for migration it might be a useful test to see if users can use null in the mapping instructions.